app/services/location_suggestion.rb

```diff
@@ -1,21 +1,34 @@
 class LocationSuggestion
+  CACHE_EXPIRY = 30.minutes
+
   class << self
     def suggest(input)
       query = build_query(input)
 
+      cached_result = Rails.cache.read(location_query_cache_key(input))
+      return cached_result if cached_result.present?
```

Comment:
Would this logic work using `cache.fetch` with a block? https://apidock.com/rails/ActiveSupport/Cache/Store/fetch. If there's an error we'll end up with `nil` in the cache anyway

Reply:
I wanted to avoid writing an entry in the cache if there's an error with the API call, but still write nil if there are genuinely no results for the query. Couldn't see how to do that with fetch.

Reply:
Ah ok. So set the key, but store nil against it? And not set the key if there was an error

Reply:
Would it make sense to store `[]` in the cache if there are no results? Otherwise you can't really distinguish between the 'no result' and 'new key' cases?

Reply:
That appeals

Reply:
@stevehook good point! It's actually an empty array that gets stored, so what needs changing here is the guard clause. Will update.

Reply:
@stevehook @duncanjbrown fyi, I've changed my approach here. It's genuinely quite difficult to get an empty result set without typing garbage into the query field. On that basis, I think this class should:
- Write a cache entry if it gets predictions back
- Do nothing in the following scenarios:
  - Server error
  - Error message in the response body (eg—API key invalid, reported in a response with status 200 😞 )
  - Empty result set

WDYT?

Reply:
Sounds good!